src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -154,6 +158,211 @@ def coe_add_monoid_hom : finite_measure α →+ measure α :=
+  [measurable_space α] (μ : measure α) [μ_fin : is_finite_measure μ]
+  {f : α → ℝ≥0∞} (f_bdd : ∃ c : ℝ≥0, ∀ x, f x ≤ c) : ∫⁻ x, f x ∂μ < ∞ :=
+begin
+  cases f_bdd with c hc,
+  apply lt_of_le_of_lt (@lintegral_mono _ _ μ _ _ hc),
+  rw lintegral_const,
+  exact ennreal.mul_lt_top ennreal.coe_lt_top.ne μ_fin.measure_univ_lt_top.ne,
+end
+
+-- Switched to using `nndist`, and believe these two could be useful... (1)
+-- Where to place?
+lemma _root_.nnreal.val_eq_nndist_zero (z : ℝ≥0) : z = nndist 0 z :=
+by { simp only [nnreal.nndist_eq, max_eq_right, zero_sub', sub_zero', zero_le'], }
+
+-- Switched to using `nndist`, and believe these two could be useful... (2)
+-- Where to place?
+lemma _root_.nnreal.val_eq_nndist_zero' (z : ℝ≥0) : z = nndist z 0 :=
+by { rw nndist_comm, exact nnreal.val_eq_nndist_zero z, }
+
+-- I think this might be occasionally useful (it is used below).
+-- Where to place?
+lemma _root_.nnreal.le_add_nndist (a b : ℝ≥0) :
+  a ≤ b + nndist a b :=
+begin
+  suffices : (a : ℝ) ≤ (b : ℝ) + (dist a b),
+  { exact nnreal.coe_le_coe.mp this,},
+  linarith [le_of_abs_le (by refl : abs (a-b : ℝ) ≤ (dist a b))],
+end
+
+-- Only useful here or more generally?
+-- Where to place?
+lemma _root_.bounded_continuous_function.nnreal.coe_comp_measurable {α : Type*}
+  [topological_space α] [measurable_space α] [opens_measurable_space α] (f : α →ᵇ ℝ≥0) :
+  measurable (λ x, (f x : ℝ≥0∞)) :=
+measurable_coe_nnreal_ennreal.comp f.continuous.measurable
+
+-- This does not seem unreasonable to me, although it may be a bit specific.
+-- Where to place?
+lemma bounded_continuous_function.nnreal.upper_bound {α : Type*} [topological_space α]
+  (f : α →ᵇ ℝ≥0) (x : α) : f x ≤ nndist f 0 :=
+begin
+  have key' : nndist (f x) ((0 : α →ᵇ ℝ≥0) x) ≤ nndist f 0,
+  { exact @bounded_continuous_function.dist_coe_le_dist α ℝ≥0 _ _ f 0 x, },
+  simp only [bounded_continuous_function.coe_zero, pi.zero_apply] at key',
+  rwa ← nnreal.val_eq_nndist_zero' (f x) at key',
+end
+
+-- This is the formulation I prefer in the present context, naturally uses the more general ones.
+lemma lintegral_lt_top_of_bounded_continuous_to_nnreal (μ : finite_measure α) (f : α →ᵇ ℝ≥0) :
+  ∫⁻ x, f x ∂(μ : measure α) < ∞ :=
+begin
+  apply is_finite_measure.lintegral_lt_top_of_bounded_to_ennreal,
+  use nndist f 0,
+  intros x,
+  have key := bounded_continuous_function.nnreal.upper_bound f x,
+  rw ennreal.coe_le_coe,
+  have eq : nndist f 0 = ⟨dist f 0, dist_nonneg⟩,
+  { ext,
+    simp only [real.coe_to_nnreal', max_eq_left_iff, subtype.coe_mk, coe_nndist], },
+  rwa eq at key,
+end
+
+lemma test_against_nn_coe_eq {μ : finite_measure α} {f : α →ᵇ nnreal} :
+  (μ.test_against_nn f : ℝ≥0∞) = ∫⁻ x, f x ∂(μ : measure α) :=
+ennreal.coe_to_nnreal (lintegral_lt_top_of_bounded_continuous_to_nnreal μ f).ne
+
+lemma test_against_nn_const (μ : finite_measure α) (c : ℝ≥0) :
+  μ.test_against_nn (bounded_continuous_function.const α c) = c * μ.mass :=
+begin
```

Comment:
I had a suspicion, which I have so far checked on the lemmas `test_against_nn_const`, `test_against_nn_mono`, `test_against_nn_add`, `test_against_nn_smul`, that the fastest way to prove facts about `test_against` is to convert to a fact about `lintegral` right at the start by rewriting with `ennreal.coe_eq_coe` or `ennreal.coe_le_coe`.  It helps if you make `test_against_nn_coe_eq` a simp-lemma and also require that `test_against_nn` take the type `α →ᵇ ℝ≥0` as an input.  The proof of this lemma then becomes
```suggestion
by simp [← ennreal.coe_eq_coe]
```
and the other three lemmas I mentioned are also pretty fast (though not quite that fast!).

Reply:
Great, thank you! I will try. This looks like a significant streamlining.

Reply:
> It helps if you make `test_against_nn_coe_eq` a simp-lemma and also require that `test_against_nn` take the type `α →ᵇ ℝ≥0` as an input.

I originally had it take `α →ᵇ ℝ≥0` as input, but based on Jason's comment <https://github.com/leanprover-community/mathlib/pull/9430#discussion_r717714330> I changed to `α → ℝ≥0`.

Mathematically I think the former makes more sense. Anyway generality is really not the purpose of `test_against_nn`, for that we have `lintegral`. Instead `test_against_nn` is used for the very specific purpose of inducing the topology. The one reason that convinced me to change was that the generality might make some compilation faster, since there would be "fewer assumptions to check". But I think `test_against_nn` should in fact be used exactly 0 times without a bundled `α →ᵇ ℝ≥0` as its input...

Now I wonder if I should change back... I think I would indeed prefer the original, except if there is a risk that the once changed version is after all preferable in mathlib...

Reply:
I haven't thought about it from the other angle at all, but I did find that the "magic simp" proofs I was suggesting for these lemmas don't work if you have it take `α → ℝ≥0` rather than `α →ᵇ ℝ≥0` as input.  The issue is that things get simped in the wrong order and the simp lemma `test_against_nn_coe_eq` doesn't get an opportunity to trigger.  For example, I think you'll find that the quick proof `by simp [← ennreal.coe_eq_coe]` I suggested for `test_against_nn_const` doesn't work if we have input type `α →ᵇ ℝ≥0` for `test_against_nn` ...?

Reply:
True, and I think this is a more compelling argument. I changed back and added `@[simp]` tags. Proofs with this idea of yours are much better.

Thank you!